src/mig/modules/memory/memory.go

```diff
@@ -0,0 +1,865 @@
+		return
+	}
+	for i, c := range s.checks {
+		if c.code&checkLib == 0 {
+			continue
+		}
+		libs, err, serr := listlibs.GetMatchingLoadedLibraries(proc, c.regex)
+		if err != nil {
+			stats.Failures = append(stats.Failures, err.Error())
+		}
+		if len(serr) > 0 && s.Options.LogFailures {
+			stats.Failures = append(stats.Failures, err.Error())
+			if debug {
+				for _, err := range serr {
+					fmt.Printf("checkLibraries: soft error -> %v\n", err)
+				}
+			}
+		}
+		if len(libs) > 0 {
+			if debug {
+				fmt.Printf("checkLibraries: proc name '%s' pid %d has libraries matching regex '%s'\n",
+					procname, proc.Pid(), c.value)
+			}
+			c.storeMatch(proc)
+		} else {
+			matchedall = false
+		}
+		s.checks[i] = c
+	}
+	return
+}
+
+func (r Runner) walkProcMemory(proc process.Process, procname string) (err error) {
+	// find longest byte string to search for, which determines the buffer size
+	bufsize := uint(4096)
+	// find lowest offset, which determines start address
+	offset := ^uintptr(0) >> 1
+	// verify that at least one search is interested in inspecting the memory
+	// of this process
+	shouldWalkMemory := false
+	// if at least one search wants to log failures, do so, otherwise omit them
+	logFailures := false
+	for label, search := range r.Parameters.Searches {
+		// if the search is not active or the search as no content or by check to run, skip it
+		if !search.isactive || (search.checkmask&checkContent == 0 && search.checkmask&checkByte == 0) {
+			search.deactivate()
+			r.Parameters.Searches[label] = search
+			continue
+		}
+		shouldWalkMemory = true
+		// find the largest bufsize needed
+		for _, c := range search.checks {
+			if c.code&checkByte != 0 {
+				if uint(len(c.bytes)) > (bufsize / 2) {
+					bufsize = 2 * uint(len(c.bytes))
+					// pad to always have an even bufsize
+					if bufsize%2 != 0 {
+						bufsize++
+					}
+				}
+			}
+		}
+		// find the smallest offset needed
+		if uintptr(search.Options.Offset) < offset {
+			offset = uintptr(search.Options.Offset)
+		}
+		if search.Options.LogFailures {
+			logFailures = true
+		}
+	}
+	if !shouldWalkMemory {
+		if debug {
+			fmt.Println("walkProcMemory: no check needs to read the memory of process", proc.Pid(), procname)
+		}
+		return
+	}
+	// keep track of the number of bytes read to exit when maxlength is reached
+	readBytes := 0
+	walkfn := func(curStartAddr uintptr, buf []byte) (keepSearching bool) {
+		if readBytes == 0 {
+			readBytes += len(buf)
+		} else {
+			readBytes += len(buf) / 2
+		}
+		if debug {
+			fmt.Println("walkProcMemory: reading", bufsize, "bytes starting at addr", curStartAddr, "; read", readBytes, "bytes so far")
+		}
+		for label, search := range r.Parameters.Searches {
+			matchedall := true
+			if !search.isactive {
+				continue
+			}
+			// if the search is meant to stop at a given address, and we're passed
+			// that point then deactivate the search now
+			if float64(readBytes) >= search.Options.MaxLength {
```

Comment:
I don't think it can overflow. The larger value readBytes could take is a uint64, and that convert to float64 fine without overflowing. But since I'm not 100% certain, I fixed it in https://github.com/mozilla/mig/commit/19c39d419ed3f37872775386876372a667dd2987